crates/proto/src/op/header.rs

```diff
@@ -82,6 +73,63 @@ pub enum MessageType {
     Response,
 }
 
+impl fmt::Display for MessageType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        let s = match self {
+            MessageType::Query => "QUERY",
+            MessageType::Response => "RESPONSE",
+        };
+
+        write!(f, "{}", s)
+    }
+}
+
+/// All the flags of the request/response header
+#[derive(Clone, Copy)]
+pub struct Flags {
+    authoritative: bool,
+    truncation: bool,
+    recursion_desired: bool,
+    recursion_available: bool,
+    authentic_data: bool,
+    checking_disabled: bool,
+}
+
+/// We are following the `dig` commands display format for the header flags
+///
+/// Example: "RD,AA,RA;" is Recursion-Desired, Authoritative-Answer, Recursion-Available.
+impl fmt::Display for Flags {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        const SEPARATOR: &str = ",";
+        let mut insert_separator = ""; // initially empty
+        if self.recursion_desired {
+            write!(f, "RD")?;
```

Comment:
I'd probably write this as a loop over `&[self.authoritative, self.truncation, self.recursion_desired, ..]` to cut down on repetition (enumerate the iterator to skip the separator on the first iteration).

Reply:
Oh, nice! I was trying to think of how to do that exactly, and that didn't occur to me... another good suggestion.